packages/next/src/cli/next-test.ts

```diff
@@ -0,0 +1,205 @@
+import { writeFileSync } from 'fs'
+import { getProjectDir } from '../lib/get-project-dir'
+import { printAndExit } from '../server/lib/utils'
+import loadConfig from '../server/config'
+import { PHASE_PRODUCTION_BUILD } from '../shared/lib/constants'
+import {
+  hasNecessaryDependencies,
+  type MissingDependency,
+} from '../lib/has-necessary-dependencies'
+import { installDependencies } from '../lib/install-dependencies'
+import type { NextConfigComplete } from '../server/config-shared'
+import findUp from 'next/dist/compiled/find-up'
+import { findPagesDir } from '../lib/find-pages-dir'
+import { verifyTypeScriptSetup } from '../lib/verify-typescript-setup'
+import path from 'path'
+import spawn from 'next/dist/compiled/cross-spawn'
+
+export interface NextTestOptions {
+  testRunner?: string
+}
+
+export const SUPPORTED_TEST_RUNNERS_LIST = ['playwright'] as const
+export type SUPPORTED_TEST_RUNNERS =
+  (typeof SUPPORTED_TEST_RUNNERS_LIST)[number]
+
+const requiredPackagesByTestRunner: {
+  [k in SUPPORTED_TEST_RUNNERS]: MissingDependency[]
+} = {
+  playwright: [
+    { file: 'playwright', pkg: '@playwright/test', exportsRestrict: false },
+  ],
+}
+
+// Type Guard for `SUPPORTED_TEST_RUNNERS` type. Unused for now, but worth keeping for future usage.
+// function isSupportedTestRunner(
+//   testRunner: string
+// ): testRunner is SUPPORTED_TEST_RUNNERS {
+//   return testRunner in requiredPackagesByTestRunner
+// }
+
+export async function nextTest(
+  directory?: string,
+  testRunnerArgs: string[] = [],
+  options: NextTestOptions = {}
+) {
+  // The following mess is in order to support an existing Next.js CLI pattern of optionally, passing a project `directory` as the first argument to execute the command on.
```

Comment:
Can we just not support this pattern instead for now until we discussed internally if we shouldn't get rid of this pattern completely? It's an experimental feature after all so I'd rather get the base functionality right without having to support each edge case.

Reply:
imo its more important to support this pattern as the rest of the CLI does and then if we can make a change to it we can come through and remove it here as well. This is gross code, but it works. Same "its experimental" mentality applies I think